OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelGpx.java

```diff
@@ -1,17 +1,136 @@
 package net.osmand.plus.wikivoyage.data;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import net.osmand.ResultMatcher;
+import net.osmand.binary.BinaryMapDataObject;
+import net.osmand.binary.BinaryMapIndexReader;
+import net.osmand.data.Amenity;
+import net.osmand.util.Algorithms;
+import net.osmand.util.MapUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.osmand.GPXUtilities.*;
+import static net.osmand.data.Amenity.REF;
+import static net.osmand.plus.helpers.GpxUiHelper.getGpxTitle;
+import static net.osmand.util.Algorithms.capitalizeFirstLetter;
+
 public class TravelGpx extends TravelArticle {
 
 	public static final String DISTANCE = "distance";
 	public static final String DIFF_ELE_UP = "diff_ele_up";
 	public static final String DIFF_ELE_DOWN = "diff_ele_down";
 	public static final String USER = "user";
 	public static final String ACTIVITY_TYPE = "route_activity_type";
+	private static final String ROUTE_TRACK_POINT = "route_track_point";
 
 	public String user;
 	public String activityType;
 	public String ref;
-	public float totalDistance = 0;
-	public double diffElevationUp = 0;
-	public double diffElevationDown = 0;
-}
+	public float totalDistance;
+	public double diffElevationUp;
+	public double diffElevationDown;
+	public int routeRadius = -1;
+
+	@Nullable
+	public GPXFile buildGpxFile(@NonNull List<BinaryMapIndexReader> readers) {
+		final List<BinaryMapDataObject> segmentList = new ArrayList<>();
+		final List<Amenity> wayPointList = new ArrayList<>();
+		for (BinaryMapIndexReader reader : readers) {
+			try {
+				if (file != null && !file.equals(reader.getFile())) {
+					continue;
+				}
+				BinaryMapIndexReader.SearchRequest<BinaryMapDataObject> sr = BinaryMapIndexReader.buildSearchRequest(
+						0, Integer.MAX_VALUE, 0, Integer.MAX_VALUE, 15, null,
+						new ResultMatcher<BinaryMapDataObject>() {
+							@Override
+							public boolean publish(BinaryMapDataObject object) {
+								if (object.getPointsLength() > 1) {
+									if (object.getTagValue(REF).equals(ref)
+											&& createTitle(object.getName()).equals(title)) {
+										segmentList.add(object);
+									}
+								}
+								return false;
+							}
+
+							@Override
+							public boolean isCancelled() {
+								return false;
+							}
+						});
+				int radiusMeters = routeRadius * 1000;
+				if (routeRadius >= 0) {
+					sr.setBBoxRadius(lat, lon, radiusMeters);
+				}
+				reader.searchMapIndex(sr);
+				BinaryMapIndexReader.SearchRequest<Amenity> pointRequest = BinaryMapIndexReader.buildSearchPoiRequest(
```

Comment:
does this line overrides 
> 
if (routeRadius &gt;= 0) {					sr.setBBoxRadius(lat, lon, radiusMeters);				}